api/v1beta1/types.go

```diff
@@ -635,6 +640,48 @@ type ServiceEndpointSpec struct {
 	Locations []string `json:"locations"`
 }
 
+// PrivateLinkServiceConnection defines the specification for a private link service connection associated with a private endpoint.
+type PrivateLinkServiceConnection struct {
+	// Name specifies the name of the private link service.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// PrivateLinkServiceID specifies the resource ID of the private link service.
+	PrivateLinkServiceID string `json:"privateLinkServiceID"`
+	// GroupIDs specifies the ID(s) of the group(s) obtained from the remote resource that this private endpoint should connect to.
+	// +optional
+	GroupIDs []string `json:"groupIDs,omitempty"`
+	// RequestMessage specifies a message passed to the owner of the remote resource with the private endpoint connection request.
+	// +kubebuilder:validation:MaxLength=140
+	// +optional
+	RequestMessage string `json:"requestMessage,omitempty"`
+}
+
+// PrivateEndpointSpec configures an Azure Private Endpoint.
+type PrivateEndpointSpec struct {
+	// Name specifies the name of the private endpoint.
+	Name string `json:"name"`
+	// Location specifies the region to create the private endpoint.
+	// +optional
+	Location string `json:"location,omitempty"`
+	// PrivateLinkServiceConnections specifies Private Link Service Connections of the private endpoint.
+	PrivateLinkServiceConnections []PrivateLinkServiceConnection `json:"privateLinkServiceConnections"`
```

Comment:
Similar to name, including `omitempty` here seems to give a slightly better error message when this is empty and doesn't need to be checked separately in the webhook:
```suggestion
	PrivateLinkServiceConnections []PrivateLinkServiceConnection `json:"privateLinkServiceConnections,omitempty"`
```
The message goes from:
```
The AzureManagedControlPlane "cluster" is invalid: spec.virtualNetwork.subnet.privateEndpoints[0].privateLinkServiceConnections: Invalid value: "null": spec.virtualNetwork.subnet.privateEndpoints[0].privateLinkServiceConnections in body must be of type array: "null"
```
to:
```
The AzureManagedControlPlane "cluster" is invalid: spec.virtualNetwork.subnet.privateEndpoints[0].privateLinkServiceConnections: Required value
```